src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs

```diff
@@ -181,7 +181,7 @@ private void AddKnownExtensions(ref IEnumerable<string> extensionPaths)
 
             if (!types.Any())
             {
-                types.AddRange(assembly.GetTypes().Where(type => type.GetTypeInfo().IsClass && !type.GetTypeInfo().IsAbstract));
+                types.AddRange(assembly.GetTypes().Where(type => type.GetTypeInfo() is { } typeInfo && typeInfo.IsClass && !typeInfo.IsAbstract));
```

Comment:
Fixes the following swallowed exception (found in the logs):

```TpTrace Warning: 0 : 71528, 1, 2022/03/16, 14:59:10.691, 5373987737093, vstest.console.dll, TestPluginDiscoverer: Failed to load extensions from file 'C:\src\vstest2\artifacts\Debug\netcoreapp2.1\Extensions\Microsoft.Diagnostics.NETCore.Client.dll'.  Skipping test extension scan for this file.  Error: System.ArgumentNullException: Value cannot be null.
Parameter name: type
   at System.Reflection.IntrospectionExtensions.GetTypeInfo(Type type)
   at Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.TestPluginDiscoverer.<>c__6`2.<GetTestExtensionsFromAssembly>b__6_1(Type type) in C:\src\vstest2\src\Microsoft.TestPlatform.Common\ExtensionFramework\TestPluginDiscoverer.cs:line 193
   at System.Linq.Enumerable.WhereArrayIterator`1.MoveNext()
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.TestPluginDiscoverer.GetTestExtensionsFromAssembly[TPluginInfo,TExtension](Assembly assembly, Dictionary`2 pluginInfos, StrinTpTrace Verbose: 0 : 68224, 1, 2022/03/16, 14:59:10.695, 5373987783709, vstest.console.exe, MetadataReaderExtensionsHelper: Discovering extensions inside assembly 'DumpMinitool, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' file path 'C:\src\vstest2\artifacts\Debug\net451\win7-x64\Extensions\DumpMinitool.exe'```

Reply:
Maybe the issue is here https://github.com/microsoft/vstest/blob/1539d8b6c2a6635d9fae371c707b8fcbac7c1be8/src/Microsoft.TestPlatform.Common/Utilities/TypesToLoadUtilities.cs#L38 looks like we can return an enumrable with something null